src/__swaps__/screens/Swap/utils/chains.ts

```diff
@@ -0,0 +1,175 @@
+  1101: 'polygonzkevm',
+  369: 'pulsechain',
+  1380012617: 'rari',
+  1918988905: 'raritestnet',
+  17001: 'redstoneholesky',
+  534352: 'scroll',
+  100: 'xdai',
+  324: 'zksync',
+};
+
+export const NATIVE_ASSETS_PER_CHAIN: Record<ChainId, AddressOrEth> = {
+  [ChainId.mainnet]: ETH_ADDRESS as Address,
+  [ChainId.hardhat]: AddressZero as Address,
+  [ChainId.goerli]: AddressZero as Address,
+  [ChainId.sepolia]: AddressZero as Address,
+  [ChainId.holesky]: AddressZero as Address,
+  [ChainId.arbitrum]: AddressZero as Address,
+  [ChainId.arbitrumGoerli]: AddressZero as Address,
+  [ChainId.arbitrumSepolia]: AddressZero as Address,
+  [ChainId.bsc]: AddressZero as Address,
+  [ChainId.bscTestnet]: AddressZero as Address,
+  [ChainId.optimism]: AddressZero as Address,
+  [ChainId.hardhatOptimism]: AddressZero as Address,
+  [ChainId.optimismSepolia]: AddressZero as Address,
+  [ChainId.rari]: AddressZero as Address,
+  [ChainId.base]: AddressZero as Address,
+  [ChainId.baseSepolia]: AddressZero as Address,
+  [ChainId.zora]: AddressZero as Address,
+  [ChainId.zoraSepolia]: AddressZero as Address,
+  [ChainId.polygon]: AddressZero as Address,
+  [ChainId.polygonMumbai]: AddressZero as Address,
+  [ChainId.avalanche]: AddressZero as Address,
+  [ChainId.avalancheFuji]: AddressZero as Address,
+  [ChainId.blast]: AddressZero as Address,
+};
+
+export const getChainName = ({ chainId }: { chainId: number }) => {
+  const network = getNetworkFromChainId(chainId);
+  return ChainNameDisplay[chainId] || network;
+};
+
+/**
+ * @desc Checks if the given chain is a Layer 2.
+ * @param chain The chain name to check.
+ * @return Whether or not the chain is an L2 network.
+ */
+export const isL2Chain = (chain: ChainName | ChainId): boolean => {
+  switch (chain) {
+    case ChainName.arbitrum:
+    case ChainName.base:
+    case ChainName.bsc:
+    case ChainName.optimism:
+    case ChainName.polygon:
+    case ChainName.zora:
+    case ChainName.avalanche:
+    case ChainId.arbitrum:
+    case ChainId.base:
+    case ChainId.bsc:
+    case ChainId.optimism:
+    case ChainId.polygon:
+    case ChainId.zora:
+    case ChainId.avalanche:
+      return true;
+    default:
+      return false;
+  }
+};
+
+export function isNativeAsset(address: AddressOrEth, chainId: ChainId) {
+  return isLowerCaseMatch(NATIVE_ASSETS_PER_CHAIN[chainId], address);
+}
+
+export function chainIdFromChainName(chainName: ChainName) {
+  return chainNameToIdMapping[chainName];
+}
+
+export function chainNameFromChainId(chainId: ChainId): ChainName {
+  return chainIdToNameMapping[chainId];
+}
+
+export const chainIdToUse = (connectedToHardhat: boolean, connectedToHardhatOp: boolean, activeSessionChainId: number) => {
+  if (connectedToHardhat) {
+    return ChainId.hardhat;
+  }
+  if (connectedToHardhatOp) {
+    return ChainId.hardhatOptimism;
+  }
+  return activeSessionChainId;
+};
+
+export const deriveChainIdByHostname = (hostname: string) => {
+  switch (hostname) {
```

Comment:
just putting it on record that this the kinda shit that i hate 😤

Reply:
agreed. luckily think we can refactor this away pretty easily